packages/adapter-pg/src/index.ts

```diff
@@ -0,0 +1,220 @@
+      return query2.rows[0];
+    },
+    async linkAccount(account) {
+      const sql = `
+      insert into accounts 
+      (
+        "userId", 
+        provider, 
+        type, 
+        "providerAccountId", 
+        access_token,
+        expires_at,
+        refresh_token,
+        id_token,
+        scope,
+        session_state,
+        token_type
+      )
+      values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)
+      returning
+        id,
+        "userId", 
+        provider, 
+        type, 
+        "providerAccountId", 
+        access_token,
+        expires_at,
+        refresh_token,
+        id_token,
+        scope,
+        session_state,
+        token_type
+      `;
+
+      const params = [
+        account.userId,
+        account.provider,
+        account.type,
+        account.providerAccountId,
+        account.access_token,
+        account.expires_at,
+        account.refresh_token,
+        account.id_token,
+        account.scope,
+        account.session_state,
+        account.token_type
+      ];
+
+      const result = await client.query(sql, params);
+      return mapExpiresAt(result.rows[0]);
+    },
+    async createSession({ sessionToken, userId, expires }) {
+      if (userId === undefined) {
+        throw Error(`userId is undef in createSession`)
+      }
+      const sql = `insert into sessions ("userId", expires, "sessionToken")
+      values ($1, $2, $3)
+      RETURNING id, "sessionToken", "userId", expires`;
+
+      const result = await client.query(sql, [userId, expires, sessionToken]);
+      return result.rows[0];
+    },
+
+    async getSessionAndUser(sessionToken: string | undefined): Promise<{
+      session: AdapterSession;
+      user: AdapterUser;
+    } | null> {
+      if (sessionToken === undefined) {
+        return null;
+      }
+      const result1 = await client.query(`select * from sessions where "sessionToken" = $1`,
+        [sessionToken]);
+      if (result1.rowCount === 0) {
+        return null
+      }
+      let session: AdapterSession = result1.rows[0];
+
+      const result2 = await client.query("select * from users where id = $1", [session.userId]);
+      if (result2.rowCount === 0) {
+        return null
+      }
+      const user = result2.rows[0];
+      return {
+        session,
+        user,
+      };
+    },
+    async updateSession(session: Partial<AdapterSession> & Pick<AdapterSession, "sessionToken">):
+      Promise<AdapterSession | null | undefined> {
+      const { sessionToken } = session;
+      const result1 = await client.query(
+        `select * from sessions where "sessionToken" = $1`
+        , [sessionToken]);
+      if (result1.rowCount === 0) {
+        return null
+      }
+      const originalSession: AdapterSession = result1.rows[0];
+
+      const newSession: AdapterSession = {
+        ...originalSession,
+        ...session
+      }
+      const sql = `
+        UPDATE sessions set
+        "userId" = $2, expires = $3
```

Comment:
we don't need to update userId since it's never passed to this function from core.